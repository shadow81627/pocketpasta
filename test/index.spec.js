// const assert = require('assert');

// describe('pocketpasta.com page', () => {
//   it('should have the right title', () => {
//     browser.url('https://pocketpasta.com');
//     const title = browser.getTitle();
//     assert.strict.equal(title, 'pocketpasta');
//   });
// });

import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuetify from 'vuetify';
import index from '@/pages/index.vue';

const localVue = createLocalVue();

localVue.use(Vuetify);

const factory = () =>
  shallowMount(index, {
    localVue,
    propsData: {
      label: 'Job Title',
      value: 'Full Stack Web Developer',
    },
  });

describe('index', () => {
  test('mounts properly', () => {
    const wrapper = factory();
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  test('renders properly', () => {
    const wrapper = factory();
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('head', () => {
    expect(index.head()).toBeTruthy();
  });
});
