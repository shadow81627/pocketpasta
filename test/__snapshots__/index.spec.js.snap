// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index renders properly 1`] = `
"<div label=\\"Job Title\\" value=\\"Full Stack Web Developer\\"><section><v-parallax-stub height=\\"600\\" alt=\\"\\" src=\\"https://res.cloudinary.com/pocketpasta/image/upload/c_fit,f_auto,q_auto,w_2200/v1559982334/hero_rko6us.jpg\\"><v-layout-stub tag=\\"div\\" column=\\"\\" align-center=\\"\\" justify-center=\\"\\" class=\\"white--text\\"><b-img-lazy-stub src=\\"https://res.cloudinary.com/pocketpasta/image/upload/c_fit,f_auto,q_auto,w_200/v1560038491/logo_vlqvdr.png\\" alt=\\"Pasta on a sheet of paper\\" width=\\"200\\" blankcolor=\\"transparent\\" offset=\\"360\\" throttle=\\"100\\"></b-img-lazy-stub> <h1 class=\\"brand white--text mb-2 text-center\\">
          PocketPasta
        </h1> <div class=\\"subheading mb-3 text-center\\">
          landing.banner.subheading
        </div> <credit-stub style=\\"position: absolute; right: 0px; bottom: 0px;\\"></credit-stub></v-layout-stub></v-parallax-stub></section> <section><v-container-stub tag=\\"div\\" grid-list-xl=\\"\\"><v-row-stub tag=\\"div\\" justify=\\"center\\" wrap=\\"\\" class=\\"my-5\\"><v-col-stub alignself=\\"center\\" tag=\\"div\\" col=\\"12\\"><v-card-stub loaderheight=\\"4\\" tag=\\"div\\" class=\\"elevation-0 transparent\\"><v-card-title-stub primarytitle=\\"\\" primary-title=\\"\\" class=\\"layout justify-center\\"><div class=\\"headline\\">landing.contact.heading</div></v-card-title-stub> <v-list-stub tag=\\"div\\" tile=\\"true\\" class=\\"transparent\\"><v-list-item-stub activeclass=\\"\\" tag=\\"div\\"><v-list-item-action-stub><v-icon-stub class=\\"text-primary\\">place</v-icon-stub></v-list-item-action-stub> <v-list-item-content-stub><v-list-item-title-stub>landing.contact.location</v-list-item-title-stub></v-list-item-content-stub></v-list-item-stub> <v-list-item-stub activeclass=\\"\\" tag=\\"div\\"><v-list-item-action-stub><v-icon-stub class=\\"text-primary\\">email</v-icon-stub></v-list-item-action-stub> <v-list-item-content-stub><v-list-item-title-stub><a href=\\"mailto:landing.contact.email\\">landing.contact.email</a></v-list-item-title-stub></v-list-item-content-stub></v-list-item-stub></v-list-stub></v-card-stub></v-col-stub></v-row-stub></v-container-stub></section></div>"
`;
